### PR TITLE
Make viper calls thread safe.

### PR DESCRIPTION
**Pull request by sramamoorthy73**
With this fix, I no longer see race errors when I perform go test -race on apidApigeeSync. It shows a bunch of unrelated races (mostly in ginkgo, and sometimes in gorilla mux). I ran go test -race for all plugin modules and ensured no race was detected for viper.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -22,39 +24,102 @@ const (
...
 		if err != nil {
 			log.Printf("Error in config file '%s': %s", configFileNameKey, err)
 		}
 
-		cfg.SetEnvPrefix("apid") // eg. env var "APID_SOMETHING" will bind to config var "something"
-		cfg.AutomaticEnv()
+		vcfg.SetEnvPrefix("apid") // eg. env var "APID_SOMETHING" will bind to config var "something"
+		vcfg.AutomaticEnv()
+
+		cfg = &ConfigMgr{vcfg: vcfg}
```
Comment (haomingzhang) [inline config/config.go:126]: This `GetConfig()` seems to be not thread-safe. If 2 threads call `GetConfig()` concurrently, `cfg` may be initialized twice.
I suggest use another lock at the beginning of `GetConfig()`, and release at last, to make `GetConfig()` a critical-section.

Comment (sramamoorthy73) [reply config/config.go:126]: I was looking into that. The reason I did not add was because go race detector did not point it. @theganyo Your opinion?

Comment (theganyo) [reply config/config.go:126]: I'd lock it.

Comment (sramamoorthy73) [reply config/config.go:126]: done

Comment (haomingzhang) [reply config/config.go:126]: Looks good to me

**Review by theganyo — APPROVED** [review]

**Pull request merged**